### DISCUSSION TITLE
What is the current state of send-email support in magit?

### DISCUSSION THREAD

**Discussion by mohkale** [magit/magit#4918] [Support] [answered]
Hi, I see currently there's not any send-email wrapper in magit. Searching for it I come upon a bunch of closed issues hinting at an eventual implementation or linking to packages that're currently archived. Could you clarify what the recommended process for running git send-email is in relation to magit? Thanks.

**Comment by tarsius** [comment, marked as answer]
> What is the current state of send-email support in magit?

Weak.

> Could you clarify what the recommended process for running git send-email is in relation to magit?

I still use `W c ...` to create patches and then I type `git send-email ...` in a terminal. This certainly could be improved, but I at least haven't gotten around to it. But others have written tooling for this task, so I would recommend looking for other packages that help with this. Maybe one day Magit will support it too, but that day isn't today.

**Reply by snan** [reply]
What does the `W c` do? I've missed out on that step!

**Reply by isadon** [reply]
@snan `W c` brings up the transient to create a patch. After that you need to either some other Emacs package or context-switch out of Emacs to send the patch via `git send-email`.

**Comment by mohkale** [comment]
I see, thanks for clarifying your workflow Tarsius. I'm curious, do you know what the main blocker for earlier implementations was? In my case I'd really just like something that wraps around notmuch and magit. Select a commit range with magit, call send-email, and then have notmuch open to let me edit it before sending off with git (or maybe with notmuch itself depending on how tough that might be to integrate :thinking:). I might try and create a little helper utility for it but would like to check if there's any prior art for this specific workflow.

**Reply by kyleam** [reply]
> In my case I'd really just like something that wraps around notmuch and magit.

In my view, that hints at the main problem: much of the work falls outside of the scope of Magit.  You're talking about integrating MUAs with Magit.

It seems like you've already looked at some past discussions, but for others interested, this comment links to a couple: https://github.com/magit/magit/issues/4028#issuecomment-572881356

**Reply by kyleam** [reply]
I also think the fact that send-email supports calling format-patch underneath complicates things.  If you're a user that wants to call send-email without a separate format-patch step, you'll see Magit's lack of a send-email transient as more of a missing piece.

In my opinion (which is [shared by Junio](https://lore.kernel.org/git/xmqqv8n3cxv9.fsf@gitster.g/T/#u), at least), it's better to prepare the patches and send them separately (i.e. use dedicated format-patch call, like Jonas mentions).  That allows you to tweak the patches and cover letter without needing to fire them off immediately and simplifies the send-email call, in most cases down to only needing the patches as arguments.
